lib/aws/client.go

```diff
@@ -0,0 +1,46 @@
+package aws
+
+import (
+	"sync"
+
+	"github.com/GetTerminus/s3helper/lib/parser"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+)
+
+type client struct {
+	sess *session.Session
+	once sync.Once
+}
+
+// Session returns an authenticated AWS session, used to make API calls.
+func (c *client) GetSession() *session.Session {
+
+	// Only ever create one session
+	c.once.Do(func() {
+		// Create a session, session.Must should handle any errors
+		// https://docs.aws.amazon.com/sdk-for-go/api/aws/session/#Must
+		c.sess = session.Must(
+			session.NewSession(
+				&aws.Config{
+					Region:      aws.String(parser.GlobalOpts.Region),
```

Comment:
I'd prefer to see these required params passed into the function rather than dug out of some global static variables. As is, this now has a temporal coupling to the parser: something in parser has to happen before this can succeed. It also obfuscates the requirements of this function since they are not clearly stated in the function signature.